rts/lra/coordinator/src/test/java/io/narayana/lra/coordinator/domain/model/LRATest.java

```diff
@@ -0,0 +1,897 @@
+                    resourcePath.getUri() + ")", 0, compensateCount.get());
+        } else {
+            /*
+             * The test is calling for a mixed outcome (a top level LRA L1 and nestedCnt nested LRAs (L2, L3, ...)::
+             * L1 the mandatory call (PUT "lraresource/multiLevelNestedActivity") registers participant C1
+             * the resource makes nestedCnt calls to "lraresource/nestedActivity" each of which create nested LRAs
+             * L2, L3, ... each of which enlists a participant (C2, C3, ...) which are completed when the call returns
+             * L2 is cancelled which causes C2 to compensate
+             * L1 is closed which triggers the completion of C1
+             *
+             * To summarise:
+             *
+             * - C1 is completed
+             * - C2 is completed and then compensated
+             * - C3, ... are completed
+             */
+
+            // compensate the first nested LRA in the enlisted resource
+            try (Response r = client.target(TestPortProvider.generateURL("/base/test/end"))
+                    .queryParam("cancel", true)
+                    .request()
+                    .header(LRA_HTTP_CONTEXT_HEADER, uris[1])
+                    .put(Entity.text(""))) {
+                assertEquals("compensate the first nested LRA", 500, r.getStatus());
+            }
+
+            lraClient.closeLRA(lra); // should not complete any nested LRAs (since they have already completed via the interceptor)
+
+            /*
+             * Expect nestedCnt + 1 completions, 1 for the top level and one for each nested LRA
+             * (NB the first nested LRA is completed and compensated)
+             * Note that the top level complete should not call complete again on the nested LRA
+             */
+            assertEquals("multiLevelNestedActivity: step 10 (called test path " +
+                    resourcePath.getUri() + ")", nestedCnt + 1, completeCount.get());
+            /*
+             * The test is calling for a mixed outcome:
+             * - the top level LRA was closed
+             * - one of the nested LRAs was compensated the rest should have been completed
+             */
+            // there should be just 1 compensation (the first nested LRA)
+            assertEquals("multiLevelNestedActivity: step 9 (called test path " +
+                    resourcePath.getUri() + ")",1, compensateCount.get());
+        }
+    }
+
+    // validate that the top level and nested LRAs are gone
+    private void assertAllFinished(URI[] uris) {
+        assertTrue(uris.length != 0);
+        IntStream.rangeClosed(0, uris.length - 1).forEach(i -> assertTrue(
+                String.format("multiLevelNestedActivity: %s LRA still active",
+                        (i == 0 ? "top level" : "nested")),
+                isFinished(uris[i])));
+    }
+
+    // This test duplicates LRACoordinatorRecovery1TestCase but using undertow instead of wildfly
```

Comment:
It says that the test is a port of LRACoordinatorRecovery1TestCase. We can remove the comment though if it is not clear or if we delete the other test. I also discussed this a little in my earlier comment above (https://github.com/jbosstm/narayana/pull/1764#discussion_r562673965)

Reply:
@ochaloup another update: Ah I missed your second question, the port of LRACoordinatorRecovery2TestCase is needed because it is unstable on our CI and was already annotated with `@Ignore`.